src/test/java/one/util/streamex/EntryStreamTest.java

```diff
@@ -153,12 +153,10 @@ public void testWithIndex() {
         // Test equals contract
         assertNotEquals(new Object(), entry);
         assertNotEquals(entry, new Object());
-        assertEquals(entry, new AbstractMap.SimpleImmutableEntry<>(0, "a"));
-    }
+        assertEquals(new AbstractMap.SimpleImmutableEntry<>(0, "a"), entry);
```

Comment:
I think it's good to test both ways, because who knows which `equals` method is called by `assertEquals`.

Reply:
For current state of `assertEquals` contract you are right.

Reply:
@amaembo But I believe that tests like these above should be a part of `ObjIntBox` (in current case) test suite.